best_practices.md

```diff
@@ -0,0 +1,147 @@
+# CLUES BEST PRACTICES
+
+A guide for getting the most out of Clues.
+
+## CTX
```

Comment:
I think both this section and the ##ERRORS section would benefit from brief blurbs about what's in the section. For this one it could be something simple like
```
The clues package can leverage golang's `context.Context` to provide additional trace information
```

Reply:
Good idea!